testproject/Assets/Tests/Runtime/MultiprocessRuntime/Helpers/BuildMultiprocessTestPlayer.cs

```diff
@@ -56,47 +56,109 @@ public static void DeleteBuild()
             }
         }
 
-        /// <summary>
-        /// Needs a separate build than the standalone test builds since we don't want the player to try to connect to the editor to do test
-        /// reporting. We only want to main node to do that, worker nodes should be dumb
-        /// </summary>
-        /// <returns></returns>
-        private static BuildReport BuildPlayer(bool isDebug = false)
+        private static BuildReport BuildPlayerUtility(BuildTarget buildTarget = BuildTarget.NoTarget, string buildPathExtension = null, bool buildDebug = false)
         {
-            // Save standalone build path to file so we can read it from standalone tests (that are not running from editor)
-            SaveBuildInfo(new BuildInfo() { BuildPath = BuildPath, IsDebug = isDebug });
+            SaveBuildInfo(new BuildInfo() { BuildPath = BuildPath });
 
             // deleting so we don't end up testing on outdated builds if there's a build failure
             DeleteBuild();
 
+            if (buildTarget == BuildTarget.NoTarget)
+            {
+                if (Application.platform == RuntimePlatform.WindowsPlayer || Application.platform == RuntimePlatform.WindowsEditor)
+                {
+                    buildPathExtension += ".exe";
+                    buildTarget = BuildTarget.StandaloneWindows64;
+                }
+                else if (Application.platform == RuntimePlatform.OSXPlayer || Application.platform == RuntimePlatform.OSXEditor)
+                {
+                    buildPathExtension += ".app";
```

Comment:
you tested this on mac correct? Cause from what I remember when testing locally on both mac and pc, build path's behaviour in Unity isn't as simple as adding an extension, folder structure will change depending on platform.

Reply:
Yes, when I launch the app from the command line I need to adjust the full path to include "Contents/MacOS/Unity" inside the ".app" folder.